ros2param/test/test_cli.py

```diff
@@ -0,0 +1,718 @@
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_set_invalid_param_hidden_node_no_hidden_argument(self):
+        with self.launch_node_command(
+                arguments=[
+                    'set', '/_hidden_param_node', 'unexistent_parameter', '1']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Node not found'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_set_invalid_param_hidden_node_hidden_argument(self):
+        with self.launch_node_command(
+                arguments=[
+                    'set', '/_hidden_param_node',
+                    'unexistent_parameter', '1', '--include-hidden-nodes']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                "Setting parameter failed: ('Invalid access to undeclared parameter(s)', [])"],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_set_inexisting_node(self):
+        with self.launch_node_command(
+                arguments=['set', '/foo/unexisting_node', 'test_param', '3.14']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Node not found'],
+            text=node_command.output,
+            strict=True
+        )
+
+    # Get verb tests without --hide-type flag
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_invalid_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'unexistent_parameter']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                'Parameter not set'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_valid_double_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'double_param']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Double value is: 1.23'],
```

Comment:
@BMarchi all of these tests use the same fixture node, and thus we have to be careful when setting and getting parameters. If `test_set_valid_param_existing_node` was run before `test_get_valid_double_param_existing_node`, this assertion would not succeed. It makes the test brittle. Consider avoiding that overlap or restoring fixture state after each test.